### DISCUSSION TITLE
Where in the specific module did the payment flow begin?

### DISCUSSION THREAD

**Discussion by Boschgopi** [juspay/hyperswitch#4087] [Q&A] [answered]
Greetings, Hyper switchers! Right now, I'm working on understanding Hyper Switch's code flow. Would you kindly let me know the precise location of the payment flow's start, meaning the module in question?

**Comment by Narayanbhat166** [comment, marked as answer]
Hey @Boschgopi, You can start by following along from the [route handlers of payments ](https://github.com/juspay/hyperswitch/blob/main/crates/router/src/routes/app.rs#L297). The main function that is called for payments is the [payments_operation_core](https://github.com/juspay/hyperswitch/blob/main/crates/router/src/core/payments.rs#L82). You can find the implementations for these functions ( get_trackers , update_trackers ) in their respective operation files, i.e For 

PaymentsCreate -> [payments_create.rs](https://github.com/juspay/hyperswitch/blob/main/crates/router/src/core/payments/operations/payment_create.rs)
PaymentsConfirm -> [payments_confirm.rs](https://github.com/juspay/hyperswitch/blob/main/crates/router/src/core/payments/operations/payment_confirm.rs)

**Reply by Boschgopi** [reply]
Many thanks for that! Narayanabhat166, Your reply is really helpful to me.
